sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/ScalaReflection.scala

```diff
@@ -329,35 +329,19 @@ object ScalaReflection extends ScalaReflection {
         }
         UnresolvedMapObjects(mapFunction, getPath, Some(cls))
 
-      case t if t <:< localTypeOf[Map[_, _]] =>
+      case t if t <:< localTypeOf[Map[_, _]] || t <:< localTypeOf[java.util.Map[_, _]] =>
```

Comment:
Alright. Should I remove the case condition modifications in this PR or leave them as they are and remove them in the next one?